crypto/evp/evp_enc.c

```diff
@@ -675,3 +978,149 @@ int EVP_CIPHER_CTX_copy(EVP_CIPHER_CTX *out, const EVP_CIPHER_CTX *in)
         }
     return 1;
 }
+
+static void *evp_cipher_from_dispatch(int nid, const OSSL_DISPATCH *fns,
+                                      OSSL_PROVIDER *prov)
+{
+    EVP_CIPHER *cipher = NULL;
+    int fnenccnt = 0, fndeccnt = 0, fnctxcnt = 0;
+
+    if ((cipher = EVP_CIPHER_meth_new(nid, 0, 0)) == NULL)
+        return NULL;
+
+    for (; fns->function_id != 0; fns++) {
+        switch (fns->function_id) {
+        case OSSL_FUNC_CIPHER_NEWCTX:
+            if (cipher->newctx != NULL)
```

Comment:
IMO, all these switch bodies read more cleanly if you invert the test:
```
if (cipher->newctx == NULL) {
    ...
}
```
But YMMV.

Reply:
I prefer to keep it as is.